modules/optimization/src/controls/ControlsReceiver.C

```diff
@@ -0,0 +1,26 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ControlsReceiver.h"
+
+registerMooseObject("OptimizationApp", ControlsReceiver);
+
+InputParameters
+ControlsReceiver::validParams()
+{
+  InputParameters params = SamplerReceiver::validParams();
+  params.addClassDescription(
+      "Control for receiving data from an Optimization Transfer.  Copy of SamplerReceiever with "
+      "OptimizationParameterTransfer added as friend class");
```

Comment:
could you just make the API you need public then use a SamplerReceiver?

Reply:
This is dangerous because we don't want any object to be able to use SamplerReceiver, only approved transfers.

Reply:
can you please elaborate on why

Reply:
This would essentially allow any object to control a parameter if it has a reference/pointer to a `SamplerReceiver` object. This leaves too much room for abuse in my opinion. An alternative is to make an interface or base class that's a friend of SamplerReceiver which has a protected function to transfer parameters.

Reply:
What would stop another object from inheriting this interface?
It's the same situation as here, once just needs to inherit both classes with new objects, add friend and off we go one can control parameters?

Reply:
done Zach removed object